File: BankingSystemAngular/src/app/service/admin/application-type.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AuthService } from '../auth.service';
import { ApplicationType } from 'src/app/Model/admin/application-type.model';

@Injectable({
  providedIn: 'root'
})
export class ApplicationTypeService {
  url: string = '';

  constructor(private httpClient: HttpClient, private auth: AuthService) {
    this.url = this.auth.getBaseUrl();
  }

  private getAuthHeaders(): HttpHeaders {
    const token = this.auth.getJwtToken();
    return new HttpHeaders({
      Authorization: `Bearer ${token}`,
    });
  }

  getAllApplicationTypes(): Observable<ApplicationType[]> {
    const headers = this.getAuthHeaders();
    return this.httpClient.get<ApplicationType[]>(`${this.url}/admin/applicationType/getAllApplicationTypes`, { headers });
  }

  addApplicationType(applicationType: string): Observable<any> {
    const headers = this.getAuthHeaders();
    return this.httpClient.post<any>(
      `${this.url}/admin/applicationType/addApplicationType?applicationType=${applicationType}`, 
      {}, 
      { headers }
    );
  }

  updateApplicationType(applicationTypeId: number, applicationType: string): Observable<any> {
    const headers = this.getAuthHeaders();
    return this.httpClient.put<any>(
      `${this.url}/admin/applicationType/updateApplicationType/${applicationTypeId}?type=${applicationType}`, 
      {}, 
      { headers }
    );
  }
  
}

File: BankingSystemSpringBoot/src/main/java/com/virtusa/springboot/BankingSystemSpringBoot/service/application/ApplicationTypeService.java
package com.virtusa.springboot.BankingSystemSpringBoot.service.application;

import java.util.List;

import com.virtusa.springboot.BankingSystemSpringBoot.model.application.ApplicationType;

public interface ApplicationTypeService {

    public ApplicationType getApplicationType(String applicationType);

    public List<ApplicationType> getApplicationTypes();

    public ApplicationType addApplicationType(String applicationType);

    public ApplicationType updateApplicationType(Long applicationTypeId, String type);

    public List<String> getAllAccountApplicationType();

    public List<String> getAllNonAccountApplicationType();
    
}
